src/main/java/org/mifos/connector/gsma/state/TransactionStateRoute.java

```diff
@@ -0,0 +1,132 @@
+                .log(LoggingLevel.INFO, "Got access token, moving on.")
+                .choice()
+                .when(exchange -> correlationIDStore.isClientCorrelationPresent(exchange.getProperty(CORRELATION_ID, String.class)))
+                    .log(LoggingLevel.INFO, "Getting Server Correlation ID")
+                    .process(exchange -> {
+                        Stream<String> serverCorrelations = correlationIDStore.getServerCorrelations(exchange.getProperty(CORRELATION_ID, String.class));
+                        String serverCorrelationID = serverCorrelations.findFirst().get();
+                        exchange.setProperty(SERVER_CORRELATION, serverCorrelationID);
+                        logger.info("Server CorrelationID: " + serverCorrelationID);
+                    })
+                    .to("direct:get-transaction-state")
+                    .log(LoggingLevel.INFO, "Transaction State API response: ${body}")
+                    .choice()
+                    .when(header("CamelHttpResponseCode").isEqualTo("200"))
+                        .to("direct:transaction-state-success")
+                    .otherwise()
+                        .log(LoggingLevel.INFO, "Transaction State Request Unsuccessful")
+                        .to("direct:transaction-state-error")
+                    .endChoice()
+                .otherwise()
+                    .log(LoggingLevel.INFO, "No Server Correlation found for Client")
+                    .to("direct:transaction-state-error")
+                .endChoice();
+
+        /**
+         * Route to call Transaction State API
+         */
+        from("direct:get-transaction-state")
+                .id("get-transaction-state")
+                .removeHeader("*")
+                .setHeader(Exchange.HTTP_METHOD, constant("GET"))
+                .setHeader("X-Date", simple(ZonedDateTime.now( ZoneOffset.UTC ).format( DateTimeFormatter.ISO_INSTANT )))
+                .setHeader("Authorization", simple("Bearer ${exchangeProperty."+ACCESS_TOKEN+"}"))
+                .toD(BaseURL + "/requeststates" + "/${exchangeProperty."+SERVER_CORRELATION+"}" + "?bridgeEndpoint=true&throwExceptionOnFailure=false");
+
+        /**
+         * Error route handler
+         * TODO: Improve based on use cases
+         */
+        from("direct:transaction-state-error")
+                .id("transaction-state-error")
+                .log(LoggingLevel.INFO, "Error in getting Transaction State")
+                .unmarshal().json(JsonLibrary.Jackson, ErrorDTO.class)
+                .process(exchange -> {
+                    exchange.setProperty(STATUS_AVAILABLE, false);
+                    exchange.setProperty(ERROR_INFORMATION, exchange.getIn().getBody(ErrorDTO.class).getErrorDescription());
+                });
+        /**
+         * Success route handler
+         */
+        from("direct:transaction-state-success")
+                .id("transaction-state-success")
```

Comment:
@SubhamPramanik : Will these route handlers work with the txn state APIs introduced in channel connector by DPC?

Reply:
Channel connector gets transaction state from the operations service, so no. During the workflow, the state will be published to Kafka, and then operations should take it up. @avikganguly01